document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/kafka/impl/ConsumerGroupHandlerImpl.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.companieshouse.document.generator.core.kafka.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.document.generator.core.kafka.ConsumerGroupHandler;
+import uk.gov.companieshouse.document.generator.core.kafka.configurationhelper.KafkaConfigHelper;
+import uk.gov.companieshouse.kafka.consumer.CHKafkaConsumerGroup;
+
+import java.util.List;
+
+@Component
+public class ConsumerGroupHandlerImpl implements ConsumerGroupHandler {
+
+    @Autowired
+    private KafkaConfigHelper kafkaConfigHelper;
+
+    /**
+     * get the consumer group
+     *
+     * @param cosumerTopics
+     * @param groupName
+     * @return
+     */
+    @Override
+    public CHKafkaConsumerGroup getConsumerGroup(List<String> cosumerTopics, String groupName) {
+        return new CHKafkaConsumerGroup(kafkaConfigHelper
+                .configureKafkaConsumer(cosumerTopics, groupName));
```

Comment:
Why is an interface needed for this helper class? We won't have more than 1 implementation for getting a consumer group.

Reply:
good practice to code to interface to decouple from there implementations, one I follow. granted there are arguments against it. And yes it was extracted to its own class to allowing mocking as we are moving away from powerMock which caused us issues.